dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java

```diff
@@ -266,33 +268,49 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
      * @param version tag for Dockstore tool
      * @param toolVersion toolVersion to return in TRS
      */
-    private static void processImageDataForToolVersion(io.dockstore.webservice.core.Tool castedContainer, Tag version,
+    public static void processImageDataForToolVersion(io.dockstore.webservice.core.Tool castedContainer, Tag version,
         ToolVersion toolVersion) {
-        ImageData data = new ImageData();
         List<Checksum> trsChecksums = new ArrayList<>();
-        if (version.getImages() != null) {
+        if (version.getImages() != null && !version.getImages().isEmpty()) {
             version.getImages().forEach(image -> {
+                ImageData data = new ImageData();
                 image.getChecksums().forEach(checksum -> {
                     Checksum trsChecksum = new Checksum();
                     trsChecksum.setType(DOCKER_IMAGE_SHA_TYPE_FOR_TRS);
                     trsChecksum.setChecksum(checksum.getChecksum());
                     trsChecksums.add(trsChecksum);
                 });
+                //TODO: for now, all container images are Docker based
+                data.setImageType(ImageType.DOCKER);
+                data.setSize(image.getSize());
+                data.setUpdated(new Date().toString());
```

Comment:
This is going to be a a locale-specific string, e.g., "Thu Jan 07 10:35:47 PST 2021" for me. I'd suggest an ISO-8601 format for this.

Reply:
Oops, this is wrong. Supposed to use the date from the image here.